torch/_inductor/select_algorithm.py

```diff
@@ -1125,215 +1125,237 @@ def __call__(
         precompilation_timeout_seconds: int = 60 * 60,
         return_multi_template=False,
     ):
-        from .codegen.cuda.cuda_kernel import CUDATemplateCaller
-
-        # Templates selected with input_gen_fns require specific input data to avoid IMA
-        # Passing custom input gen fns to benchmark_fusion NYI, so skip deferred template selection
-        # TODO(jgong5): support multi-template on CPU
-        if input_gen_fns is not None or layout.device.type == "cpu":
-            return_multi_template = False
-
-        # TODO - assert that we have not mutating kernels here
-
-        # TODO(nmacchioni): remove once CI tests are fixed
-        choices = [choice for choice in choices if choice is not None]
-
-        if mm_file_name := get_mm_log_filename():
-            M, K = input_nodes[-2].get_size()[:2]
-            N = input_nodes[-1].get_size()[-1]
-            append_to_log(mm_file_name, {"invoke": str((M, K, N))})
+        assert config.search_autotune_cache or config.max_autotune_gemm or config.max_autotune
```

Comment:
generally good check to have, we shouldn't be here if we're not doing one of these operations